Classes/Domain/Repository/DocumentRepository.php

```diff
@@ -645,4 +645,168 @@ private function findSolr($collections, $settings, $searchParams, $listedMetadat
         $search->prepare();
         return $search;
     }
+
+    /**
+     * Find the uid of the previous document relative to the current document uid.
+     * Otherwise backtrack the closest previous leaf node.
+     *
+     * @access public
+     *
+     * @param int $uid
+     *
+     * @return int|null
+     */
+    public function getPreviousDocumentUid($uid)
+    {
+        $currentDocument = $this->findOneByUid($uid);
+        $currentVolume = '';
+        $parentId = $currentDocument->getPartof();
```

Comment:
I'm not sure why this PR contains the same changes as #1380 but error is exactly the same like in previous PR.

Reply:
This PR is for the current main branch, whereas #1380 was intended for the 5.1.x release. when not required or useful we can delete it for 5.1.x.
I have added the required check, wether a `$currentDocument` was obtained from DB.